airflow/hooks/dbapi_hook.py

```diff
@@ -78,7 +79,7 @@ def get_uri(self) -> str:
         conn = self.get_connection(getattr(self, self.conn_name_attr))
         login = ''
         if conn.login:
-            login = '{conn.login}:{conn.password}@'.format(conn=conn)
+            login = '{conn.login}:{quote_plus(conn.password)}@'.format(conn=conn)
```

Comment:
Does this problem also apply to the login?

Reply:
Good point. It could. To be safe, we should apply the same to login too.

Reply:
It does -- hosted Postgresql service on Azure has an `@` in the username.